### PR TITLE
Add a checking of single spa registered apps activity functions

### PR DESCRIPTION
**Pull request by oleh-momot**

### CODE REVIEW

File: ilc/client.js [hunk 1]
```diff
@@ -95,9 +95,13 @@ document.addEventListener('click', function (e) {
         return;
     }
 
-    const href = e.target.getAttribute('href');
-    singleSpa.navigateToUrl(href);
-    e.preventDefault();
+    const pathname = e.target.getAttribute('href');
+    const singleSpaShouldNavigateToUrl = singleSpa.checkActivityFunctions({pathname}).length > 0;
```
Comment (StyleT) [inline ilc/client.js]: It doesn't work for me: https://take.ms/PoD8a

Comment (oleh-momot) [reply ilc/client.js]: Fixed

File: ilc/client.js [hunk 2]
```diff
@@ -95,9 +95,14 @@ document.addEventListener('click', function (e) {
         return;
     }
 
-    const href = e.target.getAttribute('href');
-    singleSpa.navigateToUrl(href);
-    e.preventDefault();
+    const pathname = e.target.getAttribute('href');
+    const {routeId} = router.match(pathname);
+    const isSpecialRoute = Object.values(registryConf.specialRoutes).some((route) => route.routeId === routeId);
```
Comment (StyleT) [inline ilc/client.js]: Move `isSpecialRoute` attr to result of the `router.match()`

Comment (oleh-momot) [reply ilc/client.js]: Moved

File: ilc/client.js [hunk 3]
```diff
@@ -95,9 +95,13 @@ document.addEventListener('click', function (e) {
         return;
```
Comment (StyleT) [inline ilc/client.js]: Here also check for `e.defaultPrevented === false`

Comment (oleh-momot) [reply ilc/client.js]: Updated

File: ilc/router/Router.js [hunk 4]
```diff
@@ -43,7 +45,9 @@ module.exports = class Router {
             throw new errors.NoRouteMatchError();
         }
 
-        return _defaultsDeep({}, this.#specialRoutes[404]);
+        return _defaultsDeep({
+            isSpecialRoute: true,
```
Comment (StyleT) [inline ilc/router/Router.js]: Add ability to determine type of special route

Comment (oleh-momot) [reply ilc/router/Router.js]: Updated

**Pull request merged**